tests/transforms/test_metric_tensor.py

```diff
@@ -579,6 +579,71 @@ def layer3_diag(params):
         G_expected = block_diag(G1, G3, G2)
         assert qml.math.allclose(G, G_expected, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize("interface", ["autograd", "jax", "tf", "torch"])
+    def test_argnum_metric_tensor(self, tol, interface):
+        """Test that argnum successfully reduces the number of tapes and gives
+        the desired outcome."""
+        dev = qml.device("default.qubit", wires=3)
+
+        def circuit(weights):
+            qml.RX(weights[0], wires=0)
+            qml.RY(weights[1], wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.RZ(weights[2], wires=1)
+            qml.RZ(weights[3], wires=0)
+
+        weights = np.array([0.1, 0.2, 0.3, 0.5], requires_grad=True)
+
+        with qml.tape.QuantumTape() as tape:
+            circuit(weights)
+
+        tapes, proc_fn = qml.metric_tensor(tape)
+        res = qml.execute(tapes, dev, None)
+        mt = proc_fn(res)
+        zeros = qml.math.zeros_like(mt)
+
+        tapes, proc_fn = qml.metric_tensor(tape, argnum=(0, 1, 3))
+        res = qml.execute(tapes, dev, None, interface=interface)
+        mt013 = proc_fn(res)
+
+        assert len(tapes) == 6
+        assert mt.shape == mt013.shape
+        assert qml.math.allclose(mt[:2, :2], mt013[:2, :2], atol=tol, rtol=0)
+        assert qml.math.allclose(mt[3, 3], mt013[3, 3], atol=tol, rtol=0)
+        assert qml.math.allclose(zeros[2, :], mt013[2, :], atol=tol, rtol=0)
+        assert qml.math.allclose(zeros[:, 2], mt013[:, 2], atol=tol, rtol=0)
+
+        tapes, proc_fn = qml.metric_tensor(tape, argnum=(2, 3))
+        res = qml.execute(tapes, dev, None, interface=interface)
+        mt23 = proc_fn(res)
+
+        assert len(tapes) == 1
+        assert mt.shape == mt23.shape
+        assert qml.math.allclose(mt[2:, 2:], mt23[2:, 2:], atol=tol, rtol=0)
+        assert qml.math.allclose(zeros[:2, :], mt23[:2, :], atol=tol, rtol=0)
+        assert qml.math.allclose(zeros[:, :2], mt23[:, :2], atol=tol, rtol=0)
+
+        tapes, proc_fn = qml.metric_tensor(tape, argnum=0)
+        res = qml.execute(tapes, dev, None, interface=interface)
+        mt0 = proc_fn(res)
+
+        assert len(tapes) == 1
+        assert mt.shape == mt0.shape
+        assert qml.math.allclose(mt[0, 0], mt0[0, 0], atol=tol, rtol=0)
+        assert qml.math.allclose(zeros[1:, :], mt0[1:, :], atol=tol, rtol=0)
+        assert qml.math.allclose(zeros[:, 1:], mt0[:, 1:], atol=tol, rtol=0)
+
+        warning = ''.join([
+            "Some parameters specified in argnum are not in the ",
+            "trainable parameters \[0, 1, 2, 3\] of the tape ",
+            "and will be ignored."
+        ])
+        with pytest.warns(UserWarning, match=warning):
+            tapes, proc_fn = qml.metric_tensor(tape, argnum=4)

```

Comment:
I think it would be good to put this test into a separate test function, because it does not check functionality of `argnum` but of the warning associated to it.

Reply:
Done